crates/btc-types/src/btc.rs

```diff
@@ -1501,4 +1523,35 @@ mod tests {
             "original and deserialized scriptbuf must be the same"
         );
     }
+
+    #[test]
+    fn test_bitcoin_amount_ssz_roundtrip() {
+        use ssz::{Decode, Encode};
+
+        let amount = BitcoinAmount::from_sat(12345);
+        let encoded = amount.as_ssz_bytes();
+        let decoded = BitcoinAmount::from_ssz_bytes(&encoded).unwrap();
+        assert_eq!(amount, decoded);
+    }
```

Comment:
This could be a property test `proptest!`. See https://github.com/alpenlabs/bitcoin-bosd/blob/21c257df5c45cac22b91b0c5b1ed78b9fd3dc25f/src/serde.rs#L62-L85

Check the intro in https://proptest-rs.github.io/proptest/intro.html
for a quick introduction on what are property tests if you want.